src/main/java/be/seeseemelk/mockbukkit/services/ServerBuildInfoMock.java

```diff
@@ -0,0 +1,133 @@
+package be.seeseemelk.mockbukkit.services;
+
+import be.seeseemelk.mockbukkit.MockBukkit;
+import io.papermc.paper.ServerBuildInfo;
+import net.kyori.adventure.key.Key;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.Properties;
+
+public class ServerBuildInfoMock implements ServerBuildInfo
+{
+
+	private static final Key BRAND_MOCKBUKKIT_ID = Key.key("mockbukkit:mockbukkit");
+	private final String minecraftVersion;
+	private static final String BUILD_DEV = "DEV";
+
+	public ServerBuildInfoMock()
+	{
+		try (InputStream inputStream = MockBukkit.class.getResourceAsStream("/version_info.properties"))
+		{
+			Properties properties = new Properties();
+			properties.load(inputStream);
+			this.minecraftVersion = properties.getProperty("minecraft.version");
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override
+	public @NotNull Key brandId()
+	{
+		return BRAND_MOCKBUKKIT_ID;
+	}
+
+	@Override
+	public boolean isBrandCompatible(@NotNull Key key)
+	{
+		return key.equals(BRAND_PAPER_ID) || key.equals(BRAND_MOCKBUKKIT_ID);
+	}
+
+	@Override
+	public @NotNull String brandName()
+	{
+		return "MockBukkit";
+	}
+
+	@Override
+	public @NotNull String minecraftVersionId()
+	{
+		return this.minecraftVersion;
+	}
+
+	@Override
+	public @NotNull String minecraftVersionName()
+	{
+		return this.minecraftVersion;
+	}
+
+	@Override
+	public @NotNull OptionalInt buildNumber()
+	{
+		return OptionalInt.empty();
+	}
+
+	@Override
+	public @NotNull Instant buildTime()
+	{
+		return Instant.now();
+	}
+
+	@Override
+	public @NotNull Optional<String> gitBranch()
+	{
+		return Optional.empty();
+	}
+
+	@Override
+	public @NotNull Optional<String> gitCommit()
+	{
+		return Optional.empty();
+	}
```

Comment:
If we're not going to implement these, we shouldn't implement them. We shouldn't be returning defaults.

On that note, this would also be extremely trivial to implement with the BuildParameters template class.

Reply:
I don't understand how the mockbukkit actions system works. But how would we get the branch and commit when building a new instance?

Do we need to modify actions for this?

Reply:
All you'd have to do is add new fields to Build Parameters, and set then in the blossom block in the build.gradle.kts. We already have a method or two in there was calls git, so you'd just need to add/modify them to get the info you need.

Reply:
Ah, using git from gradle is quite obvious when I think about it. Anyhow, what about the buildNumber method? What would be the mockbukkit equivalent of that; as I said before, we can nott throw unimplementedoperationexception

Reply:
We can get the build number from GitHub actions, IIRC it's the `GITHUB_RUN_NUMBER` we'd want (all variables can be found [here](https://docs.github.com/en/actions/learn-github-actions/variables)). If it's a local build then we can just return empty.